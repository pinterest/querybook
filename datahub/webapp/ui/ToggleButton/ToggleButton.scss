.ToggleButton {
    &.checked {
        .Button {
            background-color: var(--color-accent-bg);
            border-color: var(--color-accent-text);
            color: var(--color-accent-text);
        }
    }
}
