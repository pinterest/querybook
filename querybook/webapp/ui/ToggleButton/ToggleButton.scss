.ToggleButton.Button {
    &.checked {
        background-color: var(--color-accent-lightest);
        color: var(--color-accent-dark);
    }
}
