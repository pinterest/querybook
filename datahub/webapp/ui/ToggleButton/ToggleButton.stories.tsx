import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { ToggleButton } from './ToggleButton';

export const _ToggleButton = () => {
    const [checked, setChecked] = React.useState(true);
    return (
        <ToggleButton
            onClick={() => setChecked((v) => !v)}
            checked={checked}
            title={checked ? 'Toggle Button - On' : 'Toggle Button - Off'}
        />
    );
};

export default {
    title: 'Button/ToggleButton',
    decorators: [centered],
};
