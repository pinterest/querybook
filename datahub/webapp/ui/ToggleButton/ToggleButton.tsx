import React from 'react';

import './ToggleButton.scss';

interface IToggleButton {
    title: string;
    checked: boolean;
    onChange: (checked: boolean) => any;
}

export const ToggleButton: React.FunctionComponent<IToggleButton> = ({
    onChange,
    checked,
    title,
}) => {
    return (
        <div
            className={`ToggleButton ${checked ? 'checked' : ''}`}
            onClick={() => onChange(!checked)}
        >
            <span className="ToggleButton-title">{title}</span>
        </div>
    );
};
