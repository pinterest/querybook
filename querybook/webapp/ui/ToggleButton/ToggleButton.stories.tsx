import centered from '@storybook/addon-centered/react';
import React from 'react';

import { ToggleButton } from './ToggleButton';

export const _ToggleButton = () => {
    const [checked, setChecked] = React.useState(true);
    return (
        <ToggleButton
            onClick={() => setChecked((v) => !v)}
            checked={checked}
            title={checked ? 'Toggle Button - On' : 'Toggle Button - Off'}
        />
    );
};

export default {
    title: 'Button/ToggleButton',
    decorators: [centered],
};
