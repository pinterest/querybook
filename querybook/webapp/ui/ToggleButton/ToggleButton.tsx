import React from 'react';

import { Button, ButtonProps } from 'ui/Button/Button';

import './ToggleButton.scss';

export interface IToggleButtonProps extends ButtonProps {
    title: string;
    checked: boolean;
    onClick: (...args: any) => any;
}

export const ToggleButton: React.FunctionComponent<IToggleButtonProps> = ({
    onClick,
    checked,
    title,
    ...otherProps
}) => (
    <div
        className={`ToggleButton ${checked ? 'checked' : ''}`}
        onClick={() => onClick(!checked)}
    >
        <Button {...otherProps}>{title}</Button>
    </div>
);
