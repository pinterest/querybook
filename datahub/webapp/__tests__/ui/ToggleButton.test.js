import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ToggleButton } from '../../ui/ToggleButton/ToggleButton';

it('renders without crashing', () => {
    shallow(
        <ToggleButton checked={true} onChange={() => null} title="Testing" />
    );
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        let wrapper = shallow(
            <ToggleButton
                checked={true}
                onChange={() => null}
                title="Testing"
            />
        );
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot - unchecked', () => {
        let wrapper = shallow(
            <ToggleButton
                checked={false}
                onChange={() => null}
                title="Testing"
            />
        );
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});
