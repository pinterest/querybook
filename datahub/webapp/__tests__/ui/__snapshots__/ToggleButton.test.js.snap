// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot - unchecked 1`] = `
<div
  className="ToggleButton "
  onClick={[Function]}
>
  <span
    className="ToggleButton-title"
  >
    Testing
  </span>
</div>
`;

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="ToggleButton checked"
  onClick={[Function]}
>
  <span
    className="ToggleButton-title"
  >
    Testing
  </span>
</div>
`;
